### PR TITLE
Swpbl 188447

### PR DESCRIPTION
**Pull request by jehan-diaz**
The main method calls the Dynamite class functions, which will eventually own all data needed to run the app, from the application state to the UI setup. Currently we are just initializing the UI with no/blank context and new editor context on each run, so Dynamite only needs to set up the UI. The UI class owns the menu bar, central canvas, palette (which is where we will integrate Kevin's code) and the multi-purpose panel (also Kevin). 

To test: 
The edited CMake file reconfigures the build-release folder that will allow you to run the app, called Dynamite. 

When conflicts are resolved, the new and simplified main.cpp file will replace the existing one, and other files currently in Dynamite/ will be deleted.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -78,3 +78,19 @@ elseif(MSVC)
...
+add_executable(Dynamite
+    ${CMAKE_SOURCE_DIR}/imnodes.cpp
+    ${CMAKE_SOURCE_DIR}/example/Dynamite/main.cpp
+    ${CMAKE_SOURCE_DIR}/example/Dynamite/dynamite.cpp
+    ${CMAKE_SOURCE_DIR}/example/Dynamite/ui.cpp
+    ${CMAKE_SOURCE_DIR}/example/Dynamite/menubar.cpp
+    )
+target_link_libraries(Dynamite imnodes SDL2::SDL2)
+if (APPLE)
+    target_link_libraries((Dynamite) "-framework OpenGL")
```
Comment (KevinKimSonos) [inline CMakeLists.txt:91]: Need to remove the parentheses around Dynamite for Mac compatibility.
Correct to `target_link_libraries(Dynamite "-framework OpenGL")`

File: example/Dynamite/ui.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+#endif
+
+    // Create window with graphics context
+    SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
+    SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
+    SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 8);
+    SDL_WindowFlags window_flags =
+        (SDL_WindowFlags)(SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI | SDL_WINDOW_MAXIMIZED);
+    window = SDL_CreateWindow(
+        "Dear ImGui SDL2+OpenGL3 example",
```
Comment (KevinKimSonos) [inline example/Dynamite/ui.cpp:42]: The string can be changed to "Dynamite" as the application title.

File: example/Dynamite/block.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    GainFBCtrl,
+    SpeakerSizeXOverCtrl,
+    SpeakerSizeGainControl,
+    ToneHandler
+};
+
+class Block {
+
+public:
+    int id;
```
Comment (mattmooresonos) [inline example/Dynamite/block.h:46]: make these private

File: example/Dynamite/block.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    const char* name;
+    char* input;
+    char* output;
+    bool param1;
+    float param2;
+
+    bool is_active = true;
+
+    // constructors to init
+    Block(const int i, const char* n) : id(i), name(n) {}
```
Comment (mattmooresonos) [inline example/Dynamite/block.h:56]: add a default constructor with default values?

File: example/Dynamite/context.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    int start_attr, end_attr;
+};
+
+class Context {
+    public: 
+
+    ImNodesEditorContext* context = nullptr;
+    std::vector<Block>    blocks;
+    std::vector<Link>     links;
+    int                   current_id = 0;
```
Comment (mattmooresonos) [inline example/Dynamite/context.cpp:18]: hide all the member data under the private section

File: example/Dynamite/context.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+struct Link {
+    int id;
+    int start_attr, end_attr;
+};
+
+class Context {
+    public: 
+
+    ImNodesEditorContext* context = nullptr;
```
Comment (mattmooresonos) [inline example/Dynamite/context.cpp:15]: for readability please prepend all member variables with m_

File: example/Previous/GraphicalLayout.cpp [hunk 7]
```diff
@@ -0,0 +1,116 @@
+// Initial graphical layout for Project Dynamite include menu bar and canvas view
```
Comment (mattmooresonos) [inline example/Previous/GraphicalLayout.cpp:1]: can we just delete the previous folder?

**Review by mattmooresonos — APPROVED** [review]
The direction is looking good!

**Pull request merged**